examples/kitchen-sink.yaml

```diff
@@ -12,358 +12,404 @@ file_format: "0.1"
   processors:
     # Configure a batch span processor.
-    - batch:
-        # Configure delay interval (in milliseconds) between two consecutive exports.
-        #
-        # Environment variable: OTEL_BSP_SCHEDULE_DELAY
-        schedule_delay: 5000
-        # Configure maximum allowed time (in milliseconds) to export data.
-        #
-        # Environment variable: OTEL_BSP_EXPORT_TIMEOUT
-        export_timeout: 30000
-        # Configure maximum queue size.
-        #
-        # Environment variable: OTEL_BSP_MAX_QUEUE_SIZE
-        max_queue_size: 2048
-        # Configure maximum batch size.
-        #
-        # Environment variable: OTEL_BSP_MAX_EXPORT_BATCH_SIZE
-        max_export_batch_size: 512
-        # Configure exporter.
-        #
-        # Environment variable: OTEL_TRACES_EXPORTER
-        exporter:
-          # Configure exporter to be OTLP.
-          otlp:
-            # Configure protocol.
-            #
-            # Environment variable: OTEL_EXPORTER_OTLP_PROTOCOL, OTEL_EXPORTER_OTLP_TRACES_PROTOCOL
-            protocol: http/protobuf
-            # Configure endpoint.
-            #
-            # Environment variable: OTEL_EXPORTER_OTLP_ENDPOINT, OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
-            endpoint: http://localhost:4318
-            # Configure certificate.
-            #
-            # Environment variable: OTEL_EXPORTER_OTLP_CERTIFICATE, OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE
-            certificate: /app/cert.pem
-            # Configure mTLS private client key.
-            #
-            # Environment variable: OTEL_EXPORTER_OTLP_CLIENT_KEY, OTEL_EXPORTER_OTLP_TRACES_CLIENT_KEY
-            client_key: /app/cert.pem
-            # Configure mTLS client certificate.
-            #
-            # Environment variable: OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE, OTEL_EXPORTER_OTLP_TRACES_CLIENT_CERTIFICATE
-            client_certificate: /app/cert.pem
-            # Configure headers.
-            #
-            # Environment variable: OTEL_EXPORTER_OTLP_HEADERS, OTEL_EXPORTER_OTLP_TRACES_HEADERS
-            headers:
-              api-key: !!str 1234
-            # Configure compression.
-            #
-            # Environment variable: OTEL_EXPORTER_OTLP_COMPRESSION, OTEL_EXPORTER_OTLP_TRACES_COMPRESSION
-            compression: gzip
-            # Configure max time (in milliseconds) to wait for each export.
-            #
-            # Environment variable: OTEL_EXPORTER_OTLP_TIMEOUT, OTEL_EXPORTER_OTLP_TRACES_TIMEOUT
-            timeout: 10000
-            # Configure client transport security for the exporter's connection.
-            #
-            # Environment variable: OTEL_EXPORTER_OTLP_INSECURE, OTEL_EXPORTER_OTLP_TRACES_INSECURE
-            insecure: false
+    batch:
+      # Configure delay interval (in milliseconds) between two consecutive exports.
+      #
+      # Environment variable: OTEL_CONF_TRACER_PROVIDER__PROCESSORS__BATCH__SCHEDULE_DELAY
```

Comment:
wdyt of shortening the top-level config keys:
* `tracer_provider` -> `traces`
* `meter_provider` -> `metrics`
* `log_provider` -> `logs`

or if we think that's too ambiguous (e.g. tracing instrumentation could be considered part of `traces`), maybe having a top-level `sdk` config node, then the resulting env vars would be something like:
* `sdk_traces_`
* `sdk_metrics_`
* `sdk_logs_`

Reply:
The competing concerns with shortening top level keys are that on one hand we want shorter env vars, but on the other hand we want the config to mirror the SDK concepts of the specification. I.e. renaming `tracer_provider` to `traces` blurs the fact that you're configuring a tracer provider. 

> or if we think that's too ambiguous (e.g. tracing instrumentation could be considered part of traces), maybe having a top-level sdk config node, then the resulting env vars would be something like:

There's other to level attributes as well like `.attribute_limits`, `.disabled`, and `.resource`. The `sdk` disambiguates, but becomes repetitive when applied to all the properties. Begs the question of why not just nest all the top level properties under `.sdk` (i.e. `.sdk.tracer_provider`, `.sdk.meter_provider`, or `.sdk.traces`, `.sdk.metrics`)

The instrumentation question is interesting because its not clear to what extent instrumentation config will be normalized across languages, and to what extent we should even try this (if the long term vision is instrumentation moving to native, we can't force libraries to follow standard configuration).

One answer is to leave all the SDK config top level, and say that all instrumentation config is nested under `.instrumentation`, or something to that affect.